app/src/lib/git/rev-parse.ts

```diff
@@ -52,6 +52,32 @@ export async function getTopLevelWorkingDirectory(
   return Path.resolve(path, relativePath)
 }
 
+export async function checkIfRepositoryIsBare(
+  path: string
+): Promise<boolean | null> {
+  let result
+
+  try {
+    result = await git(
+      ['rev-parse', '--is-bare-repository'],
+      path,
+      'checkIfRepositoryIsBare'
+    )
+  } catch (e) {
+    if (e.message.includes('not a git repository')) {
+      return null
+    }
+
+    throw e
+  }
+
+  if (result === null) {
```

Comment:
Rather than trying to make the try-catch as narrow as possible, let's just lift this up into the try-catch to simplify things:

```
  try {
    const result = await git(
      ['rev-parse', '--is-bare-repository'],
      path,
      'checkIfRepositoryIsBare'
    )
    return result.stdout.trim() === 'true'
  } catch (e) {
    if (e.message.includes('not a git repository')) {
      return null
    }

    throw e
  }
```